zkevm-circuits/src/evm_circuit/execution/not.rs

```diff
@@ -0,0 +1,137 @@
+use crate::{
+    evm_circuit::{
+        execution::ExecutionGadget,
+        step::ExecutionState,
+        table::{FixedTableTag, Lookup},
+        util::{
+            common_gadget::SameContextGadget,
+            constraint_builder::{ConstraintBuilder, StepStateTransition, Transition::Delta},
+            CachedRegion, Word,
+        },
+        witness::{Block, Call, ExecStep, Transaction},
+    },
+    util::Expr,
+};
+use eth_types::evm_types::OpcodeId;
+use eth_types::{Field, ToLittleEndian};
+use halo2_proofs::plonk::Error;
+
+#[derive(Clone, Debug)]
+pub(crate) struct NotGadget<F> {
+    same_context: SameContextGadget<F>,
+    input: Word<F>,
+    output: Word<F>,
+}
+
+impl<F: Field> ExecutionGadget<F> for NotGadget<F> {
+    const NAME: &'static str = "NOT";
+
+    const EXECUTION_STATE: ExecutionState = ExecutionState::NOT;
+
+    fn configure(cb: &mut ConstraintBuilder<F>) -> Self {
+        let opcode = cb.query_cell();
+
+        let input = cb.query_word();
+        let output = cb.query_word();
+
+        cb.stack_pop(input.expr());
+        cb.stack_push(output.expr());
+
+        for idx in 0..32 {
+            cb.add_lookup(
+                "input XOR output is all 1's",
+                Lookup::Fixed {
+                    tag: FixedTableTag::BitwiseXor.expr(),
+                    values: [
+                        input.cells[idx].expr(),
+                        output.cells[idx].expr(),
+                        255.expr(),
+                    ],
+                },
+            );
+        }
```

Comment:
I prefer the following practice. Tiny nitpick, but IMO this is better for code maintenance.

```suggestion
        for (i, o) in input.cells.iter().zip(output.cells.iter()) { &Cell<F>, &Cell<F>                                          
            cb.add_lookup(                                                                                                      
                "input XOR output is all 1's",                                                                                  
                Lookup::Fixed {                                                                                                 
                    tag: FixedTableTag::BitwiseXor.expr(),                                                                      
                    values: [i.expr(), o.expr(), 255.expr()],                                                                   
                },                                                                                                              
            );                                                                                                                  
        }
```

Reply:
thanks! I prefer this too. I don't remember why I originally went with the idx approach.